metadata-ingestion/src/datahub/sql_parsing/sqlglot_lineage.py

```diff
@@ -1240,6 +1275,20 @@ def infer_output_schema(result: SqlParsingResult) -> Optional[List[SchemaFieldCl
     return output_schema
 
 
+def match_columns_to_schema(
+    schema_info: SchemaInfo, input_columns: List[str]
+) -> List[str]:
+    column_from_gms: List[str] = list(schema_info.keys())  # list() to silent lint
+
+    output_columns: List[str] = [
+        column
+        for column in column_from_gms
+        if column.lower() in map(str.lower, input_columns)
```

Comment:
was looking at this logic again and I don't think it's right

if a column appears in input_columns but not in the schema info, then we should leave the column name as-is. this appears to drop the column entirely

Reply:
also, we should add some unit tests for this method.

this helper method should live in schema_resolver.py, and the unit tests should go in the corresponding test file

Reply:
done